### PR TITLE
prevent unwanted copies by looping with const references

### PR DESCRIPTION
**Pull request by DubbleClick**
noticed we had a lot of these, partially consuming a lot of extra memory (plenty of std::pairs or even std::vectors), this doesn't change anything but to loop via (const) reference for performance and memory sake

may not have caught all of them (only searched for `for (auto ` and fixed manually

### CODE REVIEW

**Comment by 3vcloud** [comment]
Aren’t half of these pointers/uints anyway, so no gain?

**Comment by DubbleClick** [comment]
No, I've not made a pointer a reference (only added const in a few of those cases). Every reference added is a data struct, typically a pair, once or twice a SkillActivation struct and once even a vector.

Iterating with const ref pointers would be 'pointless' ;)

**Comment by 3vcloud** [comment]
Hoho good one! But aren’t iterators kind of pointers, you’ve just changed what the loop uses to iterate now which has the same meaning?

**Comment by DubbleClick** [comment]
Those aren't iterators though, they're std::pairs.

I don't know why half of these are called `it`

File: GWToolboxdll/Widgets/Minimap/EffectRenderer.cpp [hunk 1]
```diff
@@ -56,36 +56,36 @@ void EffectRenderer::LoadDefaults() {
...
-    for (auto triggers : aoe_effect_triggers) {
+    for (const auto& triggers : aoe_effect_triggers) {
         delete triggers.second;
     }
     aoe_effect_triggers.clear();
 }
 void EffectRenderer::Invalidate() {
     VBuffer::Invalidate();
-    for (auto p : aoe_effects) {
+    for (const auto p : aoe_effects) {
```
Comment (DubbleClick) [inline GWToolboxdll/Widgets/Minimap/EffectRenderer.cpp:70]: example of a raw pointer, no & added

File: GWToolboxdll/Widgets/Minimap/EffectRenderer.cpp [hunk 2]
```diff
@@ -56,36 +56,36 @@ void EffectRenderer::LoadDefaults() {
     aoe_effect_triggers.emplace(Spike_Trap_Activate, new EffectTrigger(Spike_Trap, 2000, GW::Constants::Range::Nearby));
 }
 EffectRenderer::~EffectRenderer() {
-    for (auto settings : aoe_effect_settings) {
+    for (const auto& settings : aoe_effect_settings) {
         delete settings.second;
     }
     aoe_effect_settings.clear();
-    for (auto triggers : aoe_effect_triggers) {
+    for (const auto& triggers : aoe_effect_triggers) {
```
Comment (DubbleClick) [inline GWToolboxdll/Widgets/Minimap/EffectRenderer.cpp:63]: example of std::pair<int, pointer>

**Review by DubbleClick — COMMENTED** [review]
can't find the std::vector atm, may have even already changed that in the other pr (which made me do a very rough search for loops here)

**Comment by 3vcloud** [comment]
Ok thats incredibly stupid of me, nice catch

**Pull request merged**